hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java

```diff
@@ -85,24 +85,17 @@
 
   private final NettyRpcClient rpcClient;
 
-  // the event loop used to set up the connection, we will also execute other operations for this
-  // connection in this event loop, to avoid locking everywhere.
-  private final EventLoop eventLoop;
```

Comment:
I'm guessing you could remove this because we made the entire `channel` synchronous? Is that right?

Reply:
Yes, that's a remove indicated by the diff.

Just a quick clarification: we didn't make the channel synchronous (at least that was not my intention, but fixme). I see 2 important changes here:
1. Call (sendRequest()) is coming on the "main" thread (or whatever the client executes). I removed the logic which marshalled the call to Netty's eventLoop thread. In other words: HBase code is running on one thread, while Netty is running on its own.
2. Netty is waiting for the connetion to be established (TCP level) before giving back the control to HBase. sync() call.